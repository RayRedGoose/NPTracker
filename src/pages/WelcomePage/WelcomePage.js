import './WelcomePage.scss'
import React from 'react'
import logo from 'assets/logo.svg'
import traveler from 'assets/traveler.svg'
import ProcessButton from 'containers/ProcessButton/ProcessButton'

const WelcomePage = () => {
  return (
    <main className="welcome">
      <img className="logo" src={logo} alt="logo"/>
      <section>
        <h1>Welcome, Traveler!</h1>
        <img src={traveler} alt="traveler"/>
        <div>
          <ProcessButton text="Sign In" type='login' />
          <ProcessButton text="Create Account" type='signup' />
        </div>
      </section>
    </main>
  )
}

export default WelcomePage
