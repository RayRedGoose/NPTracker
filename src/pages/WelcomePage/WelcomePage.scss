@import "../../variables.scss";
@import "../../index.scss";

.welcome {
  display: grid;
  grid-template-areas: "logo . ." ". main ." ". . .";
  grid-template-columns: 1fr 5fr 1fr;
  grid-template-rows: 1fr 5fr 1fr;
  &>img {
    grid-area: logo;
  }
  section {
    grid-area: main;
    text-align: center;
    &>* {
      display: block;
      margin: auto;
    }
    h1 {
      color: $accent-color-primary-dark;
      font-family: salsa;
      font-size: 64px;
      margin-bottom: 40px;
    }
    img {
      height: 300px;
      margin-bottom: 80px;
    }
  }
  button {
    margin-left: 20px;
    margin-right: 20px;
    &:first-child {
      @extend %bordered-btn;
      background: none;
    }
    &:last-child {
      @extend %accent-btn;
    }
  }
}
