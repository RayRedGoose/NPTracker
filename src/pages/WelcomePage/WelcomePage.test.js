import React from 'react';
import { shallow } from 'enzyme';
import WelcomePage from './WelcomePage';

describe('WelcomePage', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const page = shallow(<WelcomePage />);

    expect(page).toMatchSnapshot();
  });
});
