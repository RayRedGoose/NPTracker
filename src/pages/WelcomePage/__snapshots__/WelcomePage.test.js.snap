// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomePage should match the snapshot with all data passed in correctly 1`] = `
<main
  className="welcome"
>
  <img
    alt="logo"
    src="logo.svg"
  />
  <section>
    <h1>
      Welcome, Traveller!
    </h1>
    <img
      alt="traveller"
      src="traveller.svg"
    />
    <div>
      <button
        onClick={[Function]}
      >
        Sign In
      </button>
      <button
        onClick={[Function]}
      >
        Create Account
      </button>
    </div>
  </section>
</main>
`;
